alacritty_terminal/src/tty/unix.rs

```diff
@@ -167,13 +175,7 @@ pub fn new(config: &PtyConfig, window_size: WindowSize, window_id: Option<usize>
     builder.stdout(unsafe { Stdio::from_raw_fd(slave) });
 
     // Setup shell environment.
-    builder.env("LOGNAME", pw.name);
-    builder.env("USER", pw.name);
-    builder.env("HOME", pw.dir);
-
-    // Set $SHELL environment variable on macOS, since login does not do it for us.
-    #[cfg(target_os = "macos")]
-    builder.env("SHELL", config.shell.as_ref().map_or(pw.shell, Program::program));
+    builder.envs([("LOGNAME", pw.name), ("USER", pw.name), ("HOME", pw.dir)]);
```

Comment:
Please revert this change.

Reply:
Just `env` * 3 vs. `envs` part? Will do.

What are your thoughts on setting `SHELL` on macOS? My thinking right now is we shouldn't do it anywhere. It's kind of fraught, and the behavior right now results in people who set `/usr/bin/login /bin/bash` or something as their shell via `alacritty.yml` have `$SHELL` set to `/usr/bin/login`, which isn't desirable.

Reply:
Reverted `envs`. Will leave this thread open to talk about setting `SHELL`.

Reply:
Honestly don't know about `$SHELL`, clearly at some point someone somewhere thought it was necessary, but I don't really care enough about macOS to hunt down exactly the reason why.

If it works without it on the latest fresh macOS version, I'm fine with removing it. If people complain we can always add it again.

Reply:
Sounds good. Will leave it out.